app/script/event/WebSocketService.js

```diff
@@ -208,20 +214,30 @@ z.event.WebSocketService = class WebSocketService {
   sendPing() {
     const isReadyStateOpen = this.socket.readyState === 1;
     if (isReadyStateOpen) {
-      const currentTime = Date.now();
-      this.lastPingTime = this.lastPingTime || currentTime;
-      const pingIntervalDifference = this.lastPingTime - currentTime;
-
-      const maxDifference = WebSocketService.CONFIG.PING_INTERVAL + WebSocketService.CONFIG.PING_INTERVAL_THRESHOLD;
-      if (pingIntervalDifference > maxDifference) {
+      if (this._pingHasExperiencedSuspiciousInactivity()) {
         this.logger.warn('Ping interval check failed');
         return this.reconnect(WebSocketService.CHANGE_TRIGGER.PING_INTERVAL);
       }
       this.logger.info('Sending ping to WebSocket');
+      this.lastPingTime = Date.now();
```

Comment:
Is the `lastPingTime` interval also reset when `reconnect` is being called?

Reply:
No, reconnecting will not do anything with the interval.
Only a `reset` will stop the interval.
In our case, it will continue running and checking the connection every 30s.

Reply:
Ok, so after investigations, the interval is actually killed first when we reconnect then restarted with the new websocket connection.